polars/polars-time/src/windows/groupby.rs

```diff
@@ -265,9 +267,8 @@ pub(crate) fn groupby_values_iter_full_lookbehind(
             }
             last = *lower;
             i += start_offset;
-            // TODO remove unwrap once time zone is respected
-            let lower = add(&offset, *lower, NO_TIMEZONE).unwrap();
-            let upper = add(&period, lower, NO_TIMEZONE).unwrap();
+            let lower = add(tu)(&offset, *lower, tz.as_ref()).unwrap();
+            let upper = add(tu)(&period, lower, tz.as_ref()).unwrap();
```

Comment:
I couldn't figure out how to remove the `unwrap` here - if I put `.collect()` after `map`, then I get an error saying that I need a type hint, but I couldn't figure out how to do that when the result type is
```
impl Iterator<Item = (IdxSize, IdxSize)> + TrustedLen + 'a
```

Reply:
The `collect` would materialize this iterator, which is not the return type. We return a type that implements the iterator trait, so no collection. :)

Reply:
thanks for your review and for explaining!

Do you know of another way to propagate the error from here? Because `add(tu)(&offset, *lower, tz.as_ref())` could potentially land on an ambiguous or non-existent datetime, which would raise

Reply:
yup (though it would only be fallible in the tz-aware case)

OK, thanks, I'll take a look, might be a good exercise in getting better at Rust

Reply:
Yes. I'd recommend understanding the impl and dyn trait difference here. As we probably must return an enum with dynamic trait implementors here.